### PR TITLE
rkt: Do not error out when there are unrecognized lines in os-release

### PR DESCRIPTION
**Pull request by yifan-gu**
Also fix the error handling which will cause panic. Also fix the error handling which will cause panic.

cc @kubernetes/sig-rktnetes

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31022)

### CODE REVIEW

File: pkg/kubelet/rkt/rkt.go [hunk 1]
```diff
@@ -974,14 +974,14 @@ func (r *Runtime) generateRunCommand(pod *api.Pod, uuid, netnsName string) (stri
...
-	if osInfos["ID"] != "coreos" && pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.SELinuxOptions != nil {
-		runPrepared = append(runPrepared, "--no-overlay=true")
+		glog.Warningf("rkt: Failed to read the os release info: %v", err)
+	} else {
+		// Overlay fs is not supported for SELinux yet on many distros.
+		// See https://github.com/coreos/rkt/issues/1727#issuecomment-173203129.
+		// For now, coreos carries a patch to support it: https://github.com/coreos/coreos-overlay/pull/1703
+		if osInfos["ID"] != "coreos" && pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.SELinuxOptions != nil {
+			runPrepared = append(runPrepared, "--no-overlay=true")
+		}
```
Comment (tmrts) [inline pkg/kubelet/rkt/rkt.go]: whats going on here? I can't see any change, but github diff is highlighting it

Comment (tmrts) [inline pkg/kubelet/rkt/rkt.go]: nevermind I can see it clearly locally

**Comment by tmrts** [comment]
LGTM :+1:

**Comment by euank** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/kubelet/rkt/rkt.go, line 2338 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/31022#-KPZfrTDoHc772QT378v:-KPZfrTDoHc772QT378w:biqgk4r) ([raw file](https://github.com/kubernetes/kubernetes/blob/3fac36585b613706f647b24e2bee0449f85006af/pkg/kubelet/rkt/rkt.go#L2338)):*
> ```Go
> 	scanner := bufio.NewScanner(f)
> 	for scanner.Scan() {
> 		line := scanner.Text()
> ```

can we have an `if len(strings.TrimSpace(line)) == 0 continue` here? We shouldn't complain about unexpected entries in that case imo since we know at least one distro has it in a well-formed case.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/31022)*

**Comment by euank** [comment]
Thanks! One more small change to reduce the chance of logspam would be nice.

---

Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/31022#-:-KPZgVZ8G21iEgyay8i6:b-r0s96i)*

**Comment by tmrts** [comment]
@euank sorry I've missed your comment, see #31032 for the follow-up

**Pull request merged**